api/v1alpha1/jobset_types.go

```diff
@@ -74,6 +74,20 @@ type JobSetStatus struct {
 
 	// Restarts tracks the number of times the JobSet has restarted (i.e. recreated in case of RecreateAll policy).
 	Restarts int `json:"restarts,omitempty"`
+
+	// ReplicatedJobsStatus track the number of PodsReady for each replicatedJob.
+	// +optional
+	// +listType=map
+	// +listMapKey=name
+	ReplicatedJobsStatus []ReplicatedJobStatus `json:"ReplicatedJobsStatus,omitempty"`
+}
+
+// ReplicatedJobStatus defines the observed ReplicatedJobs Readiness.
+type ReplicatedJobStatus struct {
+	Name string `json:"name"`
+
+	// +kubebuilder:default=0
+	ReadyJobs int32 `json:"readyJobs"`
```

Comment:
On the Kueue side how are you comparing the number of ready jobs to the number of expected jobs for the replicatedJob? I assume you must be iterating through the `jobSet.spec.replicatedJobs` and to check the `replicas` value then comparing it with the `ReadyJobs` value in this status struct?